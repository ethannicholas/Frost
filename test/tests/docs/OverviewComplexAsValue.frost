class Complex : Value {
    def realPart:Real

    def imaginaryPart:Real

    init(realPart:Real, imaginaryPart:Real) {
        self.realPart := realPart
        self.imaginaryPart := imaginaryPart
    }

    function +(other:Complex):Complex {
        return Complex(realPart + other.realPart, imaginaryPart + other.imaginaryPart)
    }

    @override
    function get_toString():String {
        return "(\{realPart} + \{imaginaryPart}i)"
    }
}

method main() {
    def add := Complex.+
    Console.printLine(add(Complex(3, 1), Complex(7, 2))) -- prints (10 + 3i)
    def add3 := Complex(3, 0).+
    Console.printLine(add3(Complex(12, 2))) -- prints (15 + 2i)
}
