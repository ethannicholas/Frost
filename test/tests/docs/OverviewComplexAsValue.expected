(10 + 3i)
(15 + 2i)
